Bluejay.asm

```diff
@@ -3593,21 +3609,45 @@ decode_temp_done:
 	mov	Temp1, #Pgm_Beep_Strength	; Read programmed beep strength setting
 	mov	Beep_Strength, @Temp1		; Set beep strength
 
-	mov	Temp1, #Pgm_Braking_Strength	; Read programmed braking strength setting
+	; Decode pwm frequency setting
+	mov	Temp1, #Pgm_Pwm_Frequency	; Read programmed pwm frequency setting
 	mov	A, @Temp1
-IF PWM_BITS_H == 3					; Scale braking strength to pwm resolution
-	; Note: Added for completeness
+	mov	Temp2, #0					; 24 kHz by default
+	cjne	A, #48, ($+5)
+	mov	Temp2, #1					; 48 kHz
+	cjne	A, #96, ($+5)
+	mov	Temp2, #2					; 96 kHz
+
+	; TODO: Simplify?
```

Comment:
Please explain it is applying the formula:
```PWM_BITS_H EQU (2 + MCU_48MHZ - PWM_CENTERED - PWM_FREQ)```

Reply:
This I can actually explain: It basically calculates the amount of H bits which then decides the PWM mode. On BB2 this will result in a value of 1-3 and BB1 0-2. Minimum PWM mode is 8 - depending on the caluclated value it will be mode 8-10 on BB1 or 9-11 on BB2.

[Page 186 of Reference manual](https://www.keil.com/dd/docs/datashts/silabs/efm8bb2/efm8bb2-rm.pdf)

Reply:
Sorry, I didn't explain me correctly. I meant to add a comment in code to explain what it does, instead of the TODO